lib/connectors_sdk/office365/custom_client.rb

```diff
@@ -133,7 +134,7 @@ def list_items(drive_id, fields: [], break_after_page: false)
             yielded += 1
           end
 
-          if break_after_page && yielded >= 100
+          if break_after_page && yielded >= 100 && stack.any?
```

Comment:
@RossAllenBell There will be a side effect:
E.g. if a folder has more than 100 files (without amy folder), after crawling this folder completely, `yielded` will be greater than 100, but there's no more folders in the stack. Instead of breaking the page, we will move to the next drive. This is similar to incremental sync:

https://github.com/elastic/connectors/blob/c2e41de30b804b9df4e555704bd5650963a299b7/lib/connectors_sdk/office365/custom_client.rb#L172-L175

Reply:
Oh, I don't think you need that `&& stack.any?`. The scenario you describe should get handled with: https://github.com/elastic/connectors/pull/69/files#diff-f15a307a042cbd4ac669d0c8c9096c03543bab73c45d8984378a4a8bc251d56aR138.

Reply:
I guess what you are trying to do here, is to add back the current crawling folder id to the stack if it's not completed (i.e. `item_children_next_link` is present)

What I would like to prevent here is to prevent the following scenario:
1. we have crawled the drive completely, therefore there's no more folder to crawl, i.e. `stack` is empty
2. we have more than 100 items to yield

In such case, an empty stack will be saved as `page_cursor`

```
if break_after_page && yielded >= 100
  if cursors['item_children_next_link'].present?
    stack << folder_id
  end
  cursors['page_cursor'] = stack.dup
  break
end
```

Reply:
I think these changes are appropriate within this PR. The additional changes I intend to introduce should handle the scenario you describe.